src/common/http/post_file_request.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include "httpc.hpp"
+#include "unique_file_ptr.hpp"
+
+namespace http {
+class PostFile : public Request {
+public:
+    PostFile(const char *file_path, const char *url_string, const char *file_size);
```

Comment:
Passing the file size as string is… interesting. I probably see why you do that, but feels like this needs a solution.

I won't insist on solving it inside this PR.

Reply:
I completely get that it doesn't look nor feel ideal, and it was a coin toss for me again. Either it can be done this way, which allows the class to be 'clean', it could be accepted as an int and parsed to string "just in time" or parsed to a class-level char buffer.
Neither feels nice. Maybe some different way to handle headers, so that the second argument accepts whatever and parses it to strings during an actual transaction? Just a random thought.

Reply:
> parses it to strings

To clarify… you mean „render“?

I was thinking about putting some `std::variant<const char *, size_t>` into `HeaderOut`. I'm not sure how much work it would be to do that ‒ it would be cleaner, but I if it takes another day of work, then it's not worth it.

Reply:
> To clarify… you mean „render“?

The verb i use for doing something to/from strings is 'parse'. So... maybe?

I did the `std::variant` you proposed, it feels clean and it was easy to implement, so I did it.